datafusion/datasource-parquet/src/file_format.rs

```diff
@@ -1259,9 +1302,14 @@ impl FileSink for ParquetSink {
         object_store: Arc<dyn ObjectStore>,
     ) -> Result<u64> {
         let parquet_opts = &self.parquet_options;
-        let allow_single_file_parallelism =
+        let mut allow_single_file_parallelism =
             parquet_opts.global.allow_single_file_parallelism;
 
+        if parquet_opts.global.file_encryption_properties.is_some() {
+            // For now, arrow-rs does not support parallel writes with encryption
+            allow_single_file_parallelism = false;
+        }
```

Comment:
We noted this problem in the original PR and @rok is working on it. Would be nice to have a tracking issue for this @adamreeve. Noted here:
https://github.com/apache/arrow-rs/pull/7111/files#r2015196618

Reply:
There is an arrow-rs issue for this (https://github.com/apache/arrow-rs/issues/7359). Maybe we should reference that in a comment and add a todo note to remove this code once that's fixed and included in version of the parquet crate being used?